### PR TITLE
2018.3xrt

### PR DESCRIPTION
**Pull request by IshitaGhosh**
Fix for String termination error from Coverity_Fortify in  src/runtime_src/driver/hw_em/generic_pcie_hal2/debug.cxx : Feb 6th Commit hash b00fa8704a068664be916cd9455b62efc3b2ae5d

Previous commits from Jan 30, 2019 have already merged into 2018.3 as commit 544e9f18b18a2a893d68b940ee284550689ae524 using pull request #924. Please squash those old changes and review and merge the new changes from Feb 6 2019.

### CODE REVIEW

**Comment by sonals** [comment]
ok to test

File: src/runtime_src/driver/hw_em/generic_pcie_hal2/debug.cxx [hunk 1]
```diff
@@ -142,7 +142,11 @@ namespace xclhwemhal2 {
           if (count >= size) break;
           if (map->m_debug_ip_data[i].m_type == type) {
             if (baseAddress) baseAddress[count] = map->m_debug_ip_data[i].m_base_address;
-            if (portNames)   portNames[count]   = (char*)map->m_debug_ip_data[i].m_name;
+            if (portNames) {
+               char nameBuf[129] = {0};
+               std::memcpy(nameBuf, (char*)(map->m_debug_ip_data[i].m_name), 128);
+               portNames[count] = nameBuf;
+            }
```
Comment (stsoe) [inline src/runtime_src/driver/hw_em/generic_pcie_hal2/debug.cxx]: I suggest
portNames[count].assign(reinterpret_cast<const char*>(map->m_debug_ip_data[i],m_name,128);

Comment (IshitaGhosh) [reply src/runtime_src/driver/hw_em/generic_pcie_hal2/debug.cxx]: Hi Soren, I have updated the pull request with your suggested change.
Thanks for your feedback!

**Comment by stsoe** [comment]
ok to test

**Pull request merged**